doc/source/user_guide/text.rst

```diff
@@ -74,6 +74,7 @@ These are places where the behavior of ``StringDtype`` objects differ from
 l. For ``StringDtype``, :ref:`string accessor methods<api.series.str>`
    that return **numeric** output will always return a nullable integer dtype,
    rather than either int or float dtype, depending on the presence of NA values.
+   Methods returning **boolean** output will return a nullable boolean dtype.
```

Comment:
maybe add a doc-link here (can be followup)

Reply:
I'm going to hold off on this. I'm planning to restructure the docs for integer / boolean / NA once all these PRs are in.